[DOCS] AUTH-509 Reducing permissions for unauthenticated users for apiserver access (#75173)
openshift/openshift-docs: modules/unauthenticated-users-system-webhook.adoc

// Module included in the following assemblies:
//
// * cicd/builds/triggering-builds-build-hooks.adoc

:_mod-docs-content-type: PROCEDURE
[id="unauthenticated-users-system-webhook_{context}"]
= Adding unauthenticated users to the `system:webhook` role binding

> As a cluster administrator, you can add unauthenticated users to the `system:webhook` role binding in {product-title} for specific namespaces. The `system:webhook` role binding allows unauthenticated users to trigger builds from external systems that do not use an {product-title} authentication mechanism, such as GitHub, GitLab, and Bitbucket. Unauthenticated users do not have access to non-public role bindings by default. This is a change from {product-title} versions before 4.16.

Reviewer: Maybe we should use **must** here as the feature won't work without.

```suggestion
As a cluster administrator, you must add unauthenticated users to the `system:webhook` role binding in {product-title} for specific namespaces. The `system:webhook` role binding allows unauthenticated users to trigger builds from external systems that do not use an {product-title} authentication mechanism, such as GitHub, GitLab, and Bitbucket. Unauthenticated users do not have access to non-public role bindings by default. This is a change from {product-title} versions before 4.16.
```

// Module included in the following assemblies:
//
// * cicd/builds/triggering-builds-build-hooks.adoc

:_mod-docs-content-type: PROCEDURE
[id="unauthenticated-users-system-webhook_{context}"]

> = Adding unauthenticated users to the `system:webhook` role binding

Reviewer 2: Do not use markup in headings. [Doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assemblymodule-titles-and-section-headings)

As a cluster administrator, you can add unauthenticated users to the `system:webhook` role binding in {product-title} for specific namespaces. The `system:webhook` role binding allows users to trigger builds from external systems that do not use an {product-title} authentication mechanism. Unauthenticated users do not have access to non-public role bindings by default. This is a change from {product-title} versions before 4.16.

Adding unauthenticated users to the `system:webhook` role binding is required to successfully trigger builds from GitHub, GitLab, and Bitbucket.

> If it is necessary to allow unauthenticated users access to a cluster, doing so in each required namespace by adding unauthenticated users to the `system:webhook` role binding is more secure than adding unauthenticated users to the `system:webhook` cluster role binding. However, if you have a large number of namespaces, it is possible to add unauthenticated users to the `system:webhook` cluster role binding which would apply the change to all namespaces.

Reviewer 2: Feeling picky. I don't like that "doing so in each" phrase for no particular reason. Feel free to ignore. 
```suggestion
If it is necessary to allow unauthenticated users access to a cluster, you can do so by adding unauthenticated users to the `system:webhook` role binding in each required namespace. This method is more secure than adding unauthenticated users to the `system:webhook` cluster role binding. However, if you have a large number of namespaces, it is possible to add unauthenticated users to the `system:webhook` cluster role binding which would apply the change to all namespaces.
```

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

.Prerequisites

* You have access to the cluster as a user with the `cluster-admin` role.
* You have installed the OpenShift CLI (`oc`).

.Procedure

. Create a YAML file named `add-webhooks-unauth.yaml` and add the following content:
+
[source,yaml]
----
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  annotations:
    rbac.authorization.kubernetes.io/autoupdate: "true"
  name: webhook-access-unauthenticated

>   namespace: <namespace><1>

Reviewer 3: Missing space here:
```suggestion
  namespace: <namespace> <1>
```

roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: "system:webhook"
subjects:
  - apiGroup: rbac.authorization.k8s.io
    kind: Group
    name: "system:unauthenticated"
----

> <1> The namespace of your `BuildConfig`.

Reviewer 3: Just checking - should this have some sort of noun at the end?
```suggestion
<1> The namespace of your `BuildConfig` object.
```

Author: No, I don't think so. Examples: https://docs.openshift.com/container-platform/4.15/cicd/builds/basic-build-operations.html

> . Apply the configuration using the CLI:

Reviewer 3: Same comment as earlier:
```suggestion
. Apply the configuration by running the following command:
```

// Module included in the following assemblies:
//
// * cicd/builds/triggering-builds-build-hooks.adoc

:_mod-docs-content-type: PROCEDURE
[id="unauthenticated-users-system-webhook_{context}"]
= Adding unauthenticated users to the `system:webhook` role binding

As a cluster administrator, you can add unauthenticated users to the `system:webhook` role binding in {product-title} for specific namespaces. The `system:webhook` role binding allows unauthenticated users to trigger builds from external systems such as GitHub, GitLab, and Bitbucket. Unauthenticated users do not have access to private role bindings by default. This is a change from {product-title} versions before 4.16.

[IMPORTANT]
====
Always verify compliance with your organization's security standards when modifying unauthenticated access.
====

> [NOTE]

Reviewer 4: Might be unavoidable, but should try to avoid putting admonitions back-to-back (per SSG). You might be able to just put this note's text in the normal text instead of in a note. Just play with it a little and see.

====

> We recommend adding using this procedure to add the `system:webhook` role binding to each specific namespace for added security.

Reviewer 4: We don't say "we" in docs (per ISG [1]), and should also avoid "recommend" if possible (per SSG [2]). 

[1] https://www.ibm.com/docs/en/ibm-style?topic=word-usage#we
[2] https://redhat-documentation.github.io/supplementary-style-guide/#recommend

However, this statement is a little confusing, possibly contradictory. Isn't it actually decreasing security but allowing unauthenticated users access to this role binding?

So I don't think the "added security" part is right. 

Maybe you could just condense the statement to generically say to allow access to this role binding for each required namespace. And then say if you have a large number, you can use the cluster role binding.
